cluster/consensusmanager/consensusmanager.go

```diff
@@ -409,12 +440,15 @@ func (cm *ConsensusManager) handleShardEvent(event *clientv3.Event, shardPrefix
 		// Update state in memory while holding lock
 		cm.state.ShardMapping.Shards[shardID] = shardInfo
 		cm.logger.Infof("Shard %d assigned to target node %s (current: %s)", shardID, shardInfo.TargetNode, shardInfo.CurrentNode)
+		// Update shard metrics after state change
+		cm.updateShardMetrics()
```

Comment:
Do not call updateShardMetrics here. Call updateShardMetrics in Cluster.shardMappingManagementLoop when ticked.